### PR TITLE
chore(ci): add environment approval

### PR DESCRIPTION
**Pull request by ndom91**
## Reasoning 💡

Testing new `release` environment to take advantage of manually approved workflow runs.

Currently testing env only on 'release' job. Would require an approval from balazs, luis, or myself.

See https://github.com/nextauthjs/next-auth/settings/environments/257828906/edit for env settings.

## Checklist 🧢

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

### CODE REVIEW

File: .github/workflows/release.yml
Comment (lluia) [inline .github/workflows/release.yml:15]: Should we keep the `environment` approval for the `release` job? In this way, tests always run when you open a PR and get feedback straight away, but we require approval for the `release` one?

**Review by 0ubbe — COMMENTED** [review]
Looks good ✨ , small question 💭

**Comment by ndom91** [comment]
Yeah the approval in test job was literally just to test it here without having to merge, release, etc. :+1:

**Comment by ndom91** [comment]
@balazsorban44 those latest errors deploying to prod from this branch are due to these settings:

image

We can change them to these:

image

See: https://github.com/nextauthjs/next-auth/settings/environments/3486518/edit

**Comment by balazsorban44** [comment]
Thanks @ndom91! @lluia and I had a quick look at this today, and we concluded that we will probably have to create to workflows, one for PRs, and one for the release branches. So the release job is only required when we merge to certain branches. No need run the release job on PRs, as semantic release will skip release anyway

**Review by balazsorban44 — APPROVED** [review]

**Comment by balazsorban44** [comment]
@ndom91 Would like to have a call before we merge this.

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**
